src/core/blocks/core.cpp

```diff
@@ -639,6 +639,89 @@ struct PrependTo : public XpendTo {
   }
 };
 
+struct ForBlock {
+  static CBTypesInfo inputTypes() { return CoreInfo::AnyType; }
+  static CBTypesInfo outputTypes() { return CoreInfo::AnyType; }
+
+  static CBParametersInfo parameters() { return _params; }
+
+  void setParam(int index, const CBVar &value) {
+    switch (index) {
+    case 0:
+      _from = value;
+      break;
+    case 1:
+      _to = value;
+      break;
+    case 2:
+      _blocks = value;
+      break;
+    default:
+      break;
+    }
+  }
+
+  CBVar getParam(int index) {
+    switch (index) {
+    case 0:
+      return _from;
+    case 1:
+      return _to;
+    case 2:
+      return _blocks;
+    default:
+      throw InvalidParameterIndex();
+    }
+  }
+
+  void cleanup() {
+    _from.cleanup();
+    _to.cleanup();
+    _blocks.cleanup();
+  }
+
+  void warmup(CBContext *context) {
+    _from.warmup(context);
+    _to.warmup(context);
+    _blocks.warmup(context);
+  }
+
+  CBTypeInfo compose(const CBInstanceData &data) {
+    auto dataCopy = data;
+    dataCopy.inputType = CoreInfo::IntType;
+
+    _blocks.compose(dataCopy);
+
+    return data.inputType;
+  }
+
+  CBVar activate(CBContext *context, const CBVar &input) {
+    auto from = _from.get().payload.intValue;
+    auto to = _to.get().payload.intValue;
+
+    CBVar output{};
+    CBVar item{};
+    for (auto i = from; i < to; i++) {
+      item = Var(i);
+      auto state = _blocks.activate<true>(context, item, output);
+      if (state != CBChainState::Continue)
+        break;
+    }
+
+    return input;
+  }
+
+private:
+  static inline Parameters _params = {
+      {"From", CBCCSTR("TODO!"), {CoreInfo::IntType, CoreInfo::IntVarType}},
```

Comment:
Todo

Reply:
Yeah I have a bunch of todos and also I didn't implement the `help` functions.